test/ubench/mech_vec.cpp

```diff
@@ -34,173 +34,267 @@ mechanism_ptr& find_mechanism(const std::string& name, fvm_cell& cell) {
-                seg->set_compartments(num_comp_-1);
-            }
-        }
-        return std::move(c);
+        // Add soma.
+        tree.append(arb::mnpos, {{0,0,0,soma_radius}, 1});
+
+        // Add dendrite
+        tree.append(0, {{0,0,soma_radius,             dend_radius}, 3});
+        tree.append(1, {{0,0,soma_radius+dend_length, dend_radius}, 3});
+
+        d.set("soma",      tagged(1));
+        d.set("dend", tagged(3));
+
+        arb::cable_cell cell(arb::morphology(tree, true), d);
+
+        cell.paint("soma", "pas");
+        cell.paint("dend", "pas");
+
+        cell.default_parameters = arb::neuron_parameter_defaults;
+        cell.default_parameters.discretization = arb::cv_policy_max_extent(dend_length/num_comp_);
+        return std::move(cell);
     }
 
     virtual cell_kind get_cell_kind(cell_gid_type) const override {
         return cell_kind::cable;
     }
 
+    arb::util::any get_global_properties(arb::cell_kind) const override {
+        return gprop_;
+    }
 };
 
 class recipe_pas_3_branches: public recipe {
     unsigned num_comp_;
+    arb::cable_cell_global_properties gprop_;
 public:
-    recipe_pas_3_branches(unsigned num_comp): num_comp_(num_comp) {}
+    recipe_pas_3_branches(unsigned num_comp): num_comp_(num_comp) {
+        gprop_.default_parameters = arb::neuron_parameter_defaults;
+    }
 
     cell_size_type num_cells() const override {
         return 1;
     }
 
     virtual util::unique_any get_cell_description(cell_gid_type gid) const override {
-        cable_cell c;
+        using arb::reg::tagged;
 
-        auto soma = c.add_soma(12.6157/2.0);
-        soma->add_mechanism("pas");
+        arb::sample_tree tree;
+        arb::label_dict d;
 
-        c.add_cable(0, section_kind::dendrite, 1.0/2, 1.0/2, 200.0);
-        c.add_cable(1, section_kind::dendrite, 1.0/2, 1.0/2, 200.0);
-        c.add_cable(1, section_kind::dendrite, 1.0/2, 1.0/2, 200.0);
+        double soma_radius = 12.6157/2.0;
+        double dend_radius = 1.0/2;
+        double dend_length = 200;
 
-        for (auto& seg: c.segments()) {
-            if (seg->is_dendrite()) {
-                seg->add_mechanism("pas");
-                seg->set_compartments(num_comp_-1);
-            }
-        }
-        return std::move(c);
+        // Add soma.
+        tree.append(arb::mnpos, {{0,0,0,soma_radius}, 1});
+
+        // Add dendrite
+        tree.append(0, {{0          ,0          ,soma_radius,             dend_radius}, 3});
+        tree.append(1, {{0          ,0          ,soma_radius+dend_length, dend_radius}, 3});
+        tree.append(2, {{0          ,dend_length,soma_radius+dend_length, dend_radius}, 3});
+        tree.append(2, {{dend_length,0          ,soma_radius+dend_length, dend_radius}, 3});
+
+        d.set("soma",      tagged(1));
+        d.set("dend", tagged(3));
+
+        arb::cable_cell cell(arb::morphology(tree, true), d);
+
+        cell.paint("soma", "pas");
+        cell.paint("dend", "pas");
```

Comment:
Because we are applying the same mechanism to the whole cell here (and in the other recipes), you could just
```C++
cell.paint(arb::reg::all(), 'pas')
```
and initialise the cell with an empty dictionary.